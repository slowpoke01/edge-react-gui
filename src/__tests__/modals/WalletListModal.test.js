/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { WalletListModal } from '../../components/modals/WalletListModal'
import { getTheme } from '../../components/services/ThemeContext.js'
import { fakeAirshipBridgeWallet } from './TestHelpers.js'

describe('WalletListModal', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      bridge: fakeAirshipBridgeWallet,
      headerTitle: 'Wallet List',
      search: '',
      searching: true,
      theme: getTheme()
    }
    const actual = renderer.render(<WalletListModal {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
