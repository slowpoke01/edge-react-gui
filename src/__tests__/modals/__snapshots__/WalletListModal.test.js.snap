// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletListModal should render with loading props 1`] = `
<ThemedModal
  onCancel={[Function]}
>
  <ModalTitle
    center={true}
  >
    Wallet List
  </ModalTitle>
  <ForwardRef
    label="Search Wallets"
    marginRem={
      Array [
        0.5,
        0.75,
        1.25,
      ]
    }
    onBlur={[Function]}
    onChangeText={[Function]}
    onClear={[Function]}
    onFocus={[Function]}
    returnKeyType="search"
    searchIcon={true}
    value=""
  />
  <Memo(Connect(WithTheme(WalletListComponent)))
    isModal={true}
    onPress={[Function]}
    searchText=""
    searching={false}
  />
  <ModalCloseArrow
    onPress={[Function]}
  />
</ThemedModal>
`;
